openwisp_radius/utils.py

```diff
@@ -283,6 +285,104 @@ def get_group_checks(group, counters_only=False):
     return result
 
 
+def get_group_replies(group):
+    """
+    Get all RADIUS replies from the given RadiusGroup.
+
+    Args:
+        group: RadiusGroup object
+
+    Returns:
+        dict: Dictionary of RADIUS attributes with format:
+              {"attribute_name": {"op": "operator", "value": "value"}}
+              Returns empty dict if group is None
+    """
+    data = {}
+    if group:
+        for reply in group.radiusgroupreply_set.all():
+            data[reply.attribute] = {"op": reply.op, "value": reply.value}
+    return data
+
+
+def execute_counter_checks(
+    user, group, group_checks, existing_replies=None, raise_quota_exceeded=True
+):
+    """
+    Execute counter checks for all counters and return counter replies.
+
+    This function will raise MaxQuotaReached exceptions raised by counters
+    if raise_quota_exceeded is True (default); if False, it will suppress them
+    and add the counter replies to the returned dict.
+
+    Args:
+        user: User instance.
+        group: RadiusGroup instance.
+        group_checks: dict mapping check attribute names to RadiusGroupCheck objects.
+        existing_replies (dict or None): Optional existing RADIUS replies to merge with.
+            If provided, counter replies will be added/updated in this dict.
+        raise_quota_exceeded (bool): If True (default) propagate MaxQuotaReached
+            exceptions raised by counters; if False suppress them.
+
+    Returns:
+        dict: Dictionary of counter reply attributes, merged with existing_replies
+        when provided. Format is { "reply_name": remaining_value }.
+    """
+    counter_data = existing_replies or {}
+
+    for Counter in app_settings.COUNTERS:
+        if (group_check := group_checks.get(Counter.check_name)) is None:
+            continue
+        try:
+            counter = Counter(user=user, group=group, group_check=group_check)
+            remaining = counter.check()
+        except SkipCheck:
+            continue
+        except Exception as e:
+            if not isinstance(e, MaxQuotaReached):
+                logger.exception(f'Got exception "{e}" while executing {counter}')
+                continue
+            if raise_quota_exceeded:
+                raise
+
```

Comment:
Can we reduce the amount of blank lines in this PR please? Let's keep the code compact please.